junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -287,3 +287,23 @@ fun <R> assertTimeoutPreemptively(timeout: Duration, message: String, executable
 @API(status = EXPERIMENTAL, since = "5.5")
 fun <R> assertTimeoutPreemptively(timeout: Duration, message: () -> String, executable: () -> R): R =
     Assertions.assertTimeoutPreemptively(timeout, executable, message)
+
+/**
+ * Example usage:
+ * ```kotlin
+ * assertInstanceOf<RandomAccess>(list, "List should support fast random access")
+ * ```
+ * @see Assertions.assertInstanceOf
+ * @since 5.10.1
+ */
+@API(status = EXPERIMENTAL, since = "5.10.1")
+inline fun <reified T : Any> assertInstanceOf(actualValue: Any?, message: String? = null): T =
```

Comment:
It would be great for this implementation to use Kotlin's contracts feature so the instance can be used subsequently as an instance of the expected class.

```kotlin
inline fun <reified T : Any> assertInstanceOf(actualValue: Any?, message: String? = null): T {
    contract {
        returns() implies (value is T)
    }
    return Assertions.assertInstanceOf(T::class.java, actualValue, message)
}
```

will allow this pattern without cast:

```kotlin
val bar: Any = foo()
assertInstanceOf<String>(bar)
bar.lowercase() // bar is a String
```

Reply:
Let's see if it works out.